mbus/mbus-protocol.c

```diff
@@ -4015,7 +4040,7 @@ mbus_data_variable_header_xml(mbus_data_variable_header *header)
     {
         len += snprintf(&buff[len], sizeof(buff) - len, "    <SlaveInformation>\n");
 
-        len += snprintf(&buff[len], sizeof(buff) - len, "        <Id>%lld</Id>\n", mbus_data_bcd_decode(header->id_bcd, 4));
+        len += snprintf(&buff[len], sizeof(buff) - len, "        <Id>%llX</Id>\n", mbus_data_bcd_decode_hex(header->id_bcd, 4));
```

Comment:
Here and above this line you use "%llX", (upper case hexdecimal). But lines 4221 and below use "%llx" (lower case hexadecimal). 
Is this intentional, or is there a correct way that should be used all through the code?

Reply:
Oops, thanks for noticing.

Reply:
Should be fixed now